BJS2-71922

SAM19991 opened the pull request.

SAM19991 committed:
BJS2-71922

MarselYsup commented on src/main/java/school/faang/bjs271922/ListOperations.java:
```diff
@@ -0,0 +1,65 @@
+    public static int findMax(List<Integer> numbersList) {
+        return numbersList.stream()
+                .max(Integer::compareTo)
+                .orElse(0);
```
Возвращение 0 может привести к путанице, так как это может быть воспринято как реальное максимальное значение. При такой сигнатуре метода лучше выбрасывать исключение для пустого списка.

MarselYsup commented on src/main/java/school/faang/bjs271922/ListOperations.java:
```diff
@@ -0,0 +1,65 @@
+
+    public static long countStringsStartingWith(List<String> strings, char ch) {
+        return strings.stream()
+                .filter(string -> string.charAt(0) == ch)
```
стоит также проверять, что string не пустой, а то выкинется exception

MarselYsup commented on src/main/java/school/faang/bjs271922/ListOperations.java:
```diff
@@ -0,0 +1,65 @@
+                                                                String substring) {
+        return strings.stream()
+                .filter(string -> string.contains(substring))
+                .collect(Collectors.toList());
```
можно проще ```toList();```

MarselYsup commented on src/main/java/school/faang/bjs271922/ListOperations.java:
```diff
@@ -0,0 +1,65 @@
+    public static List<String> sortByLength(List<String> strings) {
+        return strings.stream()
+                .sorted(Comparator.comparingInt(String::length))
+                .collect(Collectors.toList());
```
можно проще ```toList();```

MarselYsup commented on src/main/java/school/faang/bjs271922/ListOperations.java:
```diff
@@ -0,0 +1,65 @@
+        return numbers.stream()
+                .filter(num -> num > number)
+                .min(Integer::compareTo)
+                .orElse(0);
```
тоже создаст путаницу, ведь так получается что 0 больше заданного number, а это может быть не так

```java
.orElseThrow(() -> new NoSuchElementException("Нет элементов больше %s".formatted(number)));
```

MarselYsup commented on src/main/java/school/faang/bjs271922/Main.java:
```diff
@@ -0,0 +1,38 @@
+
+public class Main {
+    public static void main(String[] args) {
+        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5, 6);
```
ожно создать последовательность чисел более лаконично:
```java
List<Integer> range = IntStream.rangeClosed(start, end).boxed().toList();
```

SAM19991 committed:
добавлен метод numbersRange(int start, int end) в ListOperations
  - генерирует список Integer через IntStream.rangeClosed().boxed().toList()

Main теперь использует ListOperations.numbersRange(1, 6) вместо Arrays.asList()

заменены Collectors.toList() на toList() в:
  - filterStringsContainingSubstring()
  - sortByLength()

countStringsStartingWith() – добавлена проверка !string.isEmpty() для пустых строк

findMinGreaterThan() – теперь бросает NoSuchElementException с осмысленным сообщением

MarselYsup commented on src/main/java/school/faang/bjs271922/ListOperations.java:
```diff
@@ -0,0 +1,72 @@
+    public static int findMax(List<Integer> numbersList) {
+        return numbersList.stream()
+                .max(Integer::compareTo)
+                .orElseThrow();
```
Стоит пробрасывать исключение с текстом ошибки

SAM19991 committed:
добавил текст об ошибке в исключение метода findMax
